azure-storage-queue/src/main/scala/akka/stream/alpakka/azure/storagequeue/FlowMapECStage.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.azure.storagequeue
+
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import scala.concurrent.{ExecutionContext, Future}
+
+/** This flow does just a map
+  * with the exception that is also hands the ExecutionContext
+  * of the materializer to the function f
```

Comment:
Yes, one could replace this with `mapAsync`.
Basically, what one would have then is that `AzureQueueSink.fromFunction` would be implemented by something like `Flow.create().mapAsync(…)` and one would have to provide an `ExecutionContext` as an additional parameter to `AzureQueueSink.fromFunction`.
This would mean that all the `apply`-functions creating Sinks would also require an `ExecutionContext` since they are implemented using `AzureQueueSink.fromFunction`.

I tried to avoid that since I have not seen that in any other api. But if you think that it is better to just use `mapAsync`, I will change that.